cst438_gradebook/src/main/java/com/cst438/controllers/AssignmentController.java

```diff
@@ -1 +1,109 @@
+package com.cst438.controllers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.cst438.domain.Assignment;
+import com.cst438.domain.AssignmentListDTO;
+import com.cst438.domain.AssignmentGrade;
+import com.cst438.domain.AssignmentGradeRepository;
+import com.cst438.domain.AssignmentRepository;
+import com.cst438.domain.Course;
+import com.cst438.domain.CourseDTOG;
+import com.cst438.domain.CourseRepository;
+import com.cst438.domain.Enrollment;
+import com.cst438.domain.GradebookDTO;
+import com.cst438.services.RegistrationService;
+
+@RestController
+public class AssignmentController {
+
+	@Autowired
+	Assignment assignment;
+	
+	@Autowired
+	AssignmentListDTO assignmentListDTO;
+	
+	@Autowired
+	AssignmentRepository assignmentRepository;
+	
+	@Autowired
+	CourseRepository courseRepository;
+	
+	@Autowired
+	RegistrationService registrationService;
+	
+	// add assignment
+	@PutMapping("/assignmentadd/{assignment}")
+	@Transactional
+	public void addAssignmentToCourse(@RequestBody Assignment assignment, @PathVariable("instructor") String instructor ) {
+		
+		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
+		
+		assignment	= checkAssignment(assignment, email);
+		assignment.setNeedsGrading(1);
+		assignmentRepository.save(assignment);
+	}
+	
+	// add assignment
+	@PostMapping("/assignmentupdate/{assignment}")

```

Comment:
I think it should be instructor

```suggestion
	@PutMapping("/assignmentadd/{instructor}")
```}")